subprojects/core/src/main/java/org/gradle/api/Project.java

```diff
@@ -1138,7 +1141,7 @@
      *
      * @param closure The closure to call.
      */
-    void beforeEvaluate(Closure closure);
+    void beforeEvaluate(@ClosureParams(value = SimpleType.class, options = {"org.gradle.api.Project"}) Closure closure);
```

Comment:
The delegate for this seems to be the build script class itself, which didn't end up being all that useful to specify here, and was also marked internal.  So, I left off the  ```@DelegatesTo```.  Same with ```afterEvaluate``` and ```container```.